driver-core/src/test/unit/com/mongodb/internal/mockito/InsufficientStubbingDetectorDemoTest.java

```diff
@@ -0,0 +1,75 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.internal.mockito;
+
+import com.mongodb.internal.binding.ReadBinding;
+import com.mongodb.internal.connection.OperationContext;
+import com.mongodb.internal.diagnostics.logging.Logger;
+import com.mongodb.internal.diagnostics.logging.Loggers;
+import com.mongodb.internal.operation.ListCollectionsOperation;
+import org.bson.BsonDocument;
+import org.bson.codecs.BsonDocumentCodec;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.mockito.internal.stubbing.answers.ThrowsException;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+final class InsufficientStubbingDetectorDemoTest {
+    private static final Logger LOGGER = Loggers.getLogger(InsufficientStubbingDetectorDemoTest.class.getSimpleName());
+
+    private ListCollectionsOperation<BsonDocument> operation;
+
+    @BeforeEach
+    void beforeEach() {
+        operation = new ListCollectionsOperation<>("db", new BsonDocumentCodec());
+    }
+
+    @Test
+    void mockObjectWithDefaultAnswer() {
+        ReadBinding binding = Mockito.mock(ReadBinding.class);
+        LOGGER.info("", assertThrows(NullPointerException.class, () -> operation.execute(binding)));
+    }
+
+    @Test
+    void mockObjectWithThrowsException() {
+        ReadBinding binding = Mockito.mock(ReadBinding.class,
+                new ThrowsException(new AssertionError("Insufficient stubbing for " + ReadBinding.class)));
+        LOGGER.info("", assertThrows(AssertionError.class, () -> operation.execute(binding)));
+    }
+
+    @Test
+    void mockObjectWithInsufficientStubbingDetector() {
+        ReadBinding binding = MongoMockito.mock(ReadBinding.class);
+        LOGGER.info("", assertThrows(AssertionError.class, () -> operation.execute(binding)));
+    }
+
+    @Test
+    void stubbingWithThrowsException() {
+        ReadBinding binding = Mockito.mock(ReadBinding.class,
+                new ThrowsException(new AssertionError("Unfortunately, you cannot do stubbing")));
+        assertThrows(AssertionError.class, () -> when(binding.getOperationContext()).thenReturn(new OperationContext()));
+    }
+
+    @Test
+    void stubbingWithInsufficientStubbingDetector() {
+        MongoMockito.mock(ReadBinding.class, bindingMock ->
+                when(bindingMock.getOperationContext()).thenReturn(new OperationContext())
+        );
+    }
```

Comment:
For some reason, unused Mockito stubbing is not reported when we run tests (this has nothing to do with `MongoMockito`, as I tried the same with `Mockito`, and nothing was reported there either). If someone knows why, please comment.

Reply:
Add the juniper extension and 💥 

```@ExtendWith(MockitoExtension.class)```

Reply:
Indeed, thank you! After looking at its code, I discovered that the following manually written code also enables validation:

```java
private MockitoSession session;

 @BeforeEach
 void beforeEach() {
     session = Mockito.mockitoSession().startMocking();
 }
 @AfterEach
 void afterEach() {
     session.finishMocking();
 }
```

`finishMocking` calls `Mockito.validateMockitoUsage`, but only after notifying `Mockito.framework()` that a test has finished via `UniversalTestListener.testFinished`.